### PR TITLE
Fix journalctl to work with persistent logs

### PR DESCRIPTION
**Pull request by pipex**
v14.4.5 introduced a new way to retrieve the host logs by symlinking the log directories into the host /run/log/journal and using a local copy of the journalctl binary instead of chrooting into the host. This failed to account for the location of persistent logs when that option is turned on (/var/log/journal instead of /run/log/journal). This fixes the checks consider this case and also makes sure to remove the original symlink to avoid having nested links `/run/log/journal/journal/journal`, which was a bug in the previous implementation.

Change-type: patch

### CODE REVIEW

File: entry.sh [hunk 1]
```diff
@@ -30,10 +30,22 @@ fi
 # NOTE: this won't be necessary once the supervisor can update
 # itself, as using the label io.balena.features.journal-logs will
 # achieve the same objective
-if [ -d /mnt/root/run/log/journal ]; then
+if { [ ! -d /run/log/journal ] || [ -L /run/log/journal ]; } && [ -s /mnt/root/etc/machine-id ]; then
```
Comment (cywang117) [inline entry.sh:33]:
```suggestion
if { [ ! -d /run/log/journal ] || [ ! -L /run/log/journal ]; } && [ -s /mnt/root/etc/machine-id ]; then
```
Not a symlink --> forgot the `!`

Comment (cywang117) [reply entry.sh:33]: Otherwise LGTM!

Comment (pipex) [reply entry.sh:33]: Sorry, my comment in the line below is wrong. `-d` returns true even if the file is a symlink. If the directory exists, then only if the file is a symlink we want to enter the `if` clause. I'll fix the comment

Comment (pipex) [reply entry.sh:33]: No, my comment  comment in the following line is wrong. If the directory exists, then we only want to enter the `if` section if the file is a symlink (`-d` returns true

Comment (pipex) [reply entry.sh:33]: I updated the comment

**Comment by pipex** [comment]
@balena-ci I self certify!

**Pull request merged**